lib/std/jsonutils.nim

```diff
@@ -92,31 +90,27 @@ proc checkJsonImpl(cond: bool, condStr: string, msg = "") =
 template checkJson(cond: untyped, msg = "") =
   checkJsonImpl(cond, astToStr(cond), msg)
 
-template fromJsonFields(a, b, T, keys) =
+template fromJsonFields(a, b, T, keys, allowMissingFields) =
   checkJson b.kind == JObject, $(b.kind) # we could customize whether to allow JNull
   var num = 0
   for key, val in fieldPairs(a):
     num.inc
     when key notin keys:
       if b.hasKey key:
         fromJson(val, b[key])
-      else:
-        # we could customize to allow this
+      elif not allowMissingFields:
         checkJson false, $($T, key, b)
-  checkJson b.len == num, $(b.len, num, $T, b) # could customize
+  if not allowMissingFields:
+    checkJson b.len == num, $(b.len, num, $T, b)
 
-proc fromJson*[T](a: var T, b: JsonNode) =
+proc fromJson*[T](a: var T, b: JsonNode, allowMissingFields = false) =
```

Comment:
add a test in tjsonutils with and without `allowMissingFields`

Reply:
Done.